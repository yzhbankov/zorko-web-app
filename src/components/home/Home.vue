<template>
    <v-layout>
      <v-flex xs12>
        <v-container fluid v-bind="{ [`grid-list-${size}`]: true }">
          <v-layout row wrap>
            <v-flex
              v-if="looksPreview"
              xs4
              v-for="look in popularLooks"
              :key="look.title"
            >
              <HomeLooksPreview
                :src="look.preview"
                :title="look.title"
                :path="look.path"
                :author="look.author"
                :datum="look.datum"
                @openLook="onOpenLook"
                @openDatum="onOpenDatum"
              />
            </v-flex>
            <v-flex
              v-if="datumsPreview"
              xs12>
              <v-list two-line subheader>
                <HomeDatumsPreview
                  v-for="datum in popularDatums"
                  :name="datum.name"
                  :title="datum.title"
                  :connection="datum.connection"
                  :description="datum.description"
                  :info="datum.info"
                  :looksCount="datum.looksCount"
                  @openDatum="onOpenDatum"
                />
              </v-list>
            </v-flex>

          </v-layout>
        </v-container>
      </v-flex>
    </v-layout>
</template>

<script>
  import HomeLooksPreview from './HomeLooksPreview.vue'
  import HomeDatumsPreview from './HomeDatumsPreview.vue'

  import { createNamespacedHelpers } from 'vuex'
  const { mapGetters } = createNamespacedHelpers('home/')

  export default {
    name: 'Home',
    components: {
      HomeLooksPreview,
      HomeDatumsPreview
    },
    props: {
      looksPreview: {
        type: Boolean
      },
      datumsPreview: {
        type: Boolean
      }
    },
    computed: {
      ...mapGetters([
        'popularLooks',
        'popularDatums'
      ])
    },
    methods: {
      onOpenLook (item) {
        this.$router.push(item.path)
      },
      onOpenDatum ({name}) {
        this.$router.push('/datums/' + name)
      }
    },
    data () {
      return {
        size: 'lg'
      }
    }
  }
</script>
