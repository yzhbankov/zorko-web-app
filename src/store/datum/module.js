import state from './state'
import * as mutations from './mutations'
import * as actions from './actions'

export default ({namespaced}) => ({
  namespaced,
  state: state(),
  mutations,
  actions
})
