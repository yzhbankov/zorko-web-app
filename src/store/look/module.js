import state from './state'
import * as actions from './actions'
import * as mutations from './mutations'

export default ({namespaced}) => ({
  namespaced,
  state: state(),
  actions,
  mutations
})
