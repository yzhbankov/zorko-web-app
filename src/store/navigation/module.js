import state from './state'
import * as mutations from './mutations'

export default ({namespaced}) => ({
  namespaced,
  state: state(),
  mutations
})
