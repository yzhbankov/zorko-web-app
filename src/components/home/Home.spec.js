import {createLocalVue, mount, shallow} from 'vue-test-utils'
import Vuex from 'vuex'
import Home from './Home.vue'
import module from '../../store/home/module'

const localVue = createLocalVue()
const homeModule = module({namespaced: true})

localVue.use(Vuex)

describe('Home.vue', () => {
  let store

  beforeEach(() => {
    store = new Vuex.Store({
      state: {},
      modules: {
        home: {
          getters: homeModule.getters,
          state: homeModule.state
        }
      }
    })
  })

  it('initialize', () => {
    const wrapper = mount(Home, {store, localVue})

    expect(wrapper.vm).toBeTruthy()
  })

  it('redirects to look if look opened', () => {
    const lookInfo = {title: 'blblbl', path: 'bazzz'}
    const $router = {
      push: jest.fn()
    }
    const wrapper = shallow(Home, {
      store,
      localVue,
      mocks: {
        $router
      }
    })

    wrapper.vm.onOpenLook(lookInfo)

    expect($router.push).toHaveBeenCalledWith(lookInfo.path)
  })

  it('redirects to datum if datum opened', () => {
    const datumInfo = {name: 'bazzz'}
    const $router = {
      push: jest.fn()
    }
    const wrapper = shallow(Home, {
      store,
      localVue,
      mocks: {
        $router
      }
    })

    wrapper.vm.onOpenDatum(datumInfo)

    expect($router.push).toHaveBeenCalledWith('/datums/bazzz')
  })
})
