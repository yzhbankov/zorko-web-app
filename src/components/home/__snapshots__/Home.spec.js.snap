// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue has same HTML for datums preview 1`] = `
<v-layout>
    <v-flex xs12="">
        <v-container fluid="" grid-list-lg="true">
            <v-layout row="" wrap="">
                <!---->
                <!---->
                <!---->
                <!---->
                <!---->
                <v-flex xs12="">
                    <v-list two-line="" subheader="">
                        <!---->
                        <!---->
                    </v-list>
                </v-flex>
            </v-layout>
        </v-container>
    </v-flex>
</v-layout>
`;

exports[`Home.vue has same HTML for looks preview 1`] = `
<v-layout>
    <v-flex xs12="">
        <v-container fluid="" grid-list-lg="true">
            <v-layout row="" wrap="">
                <v-flex xs4="">
                    <!---->
                </v-flex>
                <v-flex xs4="">
                    <!---->
                </v-flex>
                <v-flex xs4="">
                    <!---->
                </v-flex>
                <v-flex xs4="">
                    <!---->
                </v-flex>
                <v-flex xs4="">
                    <!---->
                </v-flex>
                <!---->
            </v-layout>
        </v-container>
    </v-flex>
</v-layout>
`;
